agent/client/workflow_listen_for_poll_requests.go

```diff
@@ -10,20 +10,27 @@ import (
 )
 
 func (c *Client) startPollerListener() error {
-	ctx := context.Background()
+	ctx, cancelCtx := context.WithCancel(context.Background())
```

Comment:
I cannot `defer cancelCtx` here because this function ends while the goroutine keeps going on. If I defer the call, the goroutine will break.

Reply:
Can you add a comment explaining this situation on the code?